tests/test_evpn_tunnel.py

```diff
@@ -0,0 +1,749 @@
+    entries = set(tbl.getKeys())
+    new_entries = list(entries - existed_entries)
+    assert len(new_entries) == count, "Wrong number of created entries."
+    new_entries.sort()
+    return new_entries
+
+def get_default_vr_id(dvs):
+    db = swsscommon.DBConnector(swsscommon.ASIC_DB, dvs.redis_sock, 0)
+    table = 'ASIC_STATE:SAI_OBJECT_TYPE_VIRTUAL_ROUTER'
+    tbl =  swsscommon.Table(db, table)
+    keys = tbl.getKeys()
+    assert len(keys) == 1, "Wrong number of virtual routers found"
+
+    return keys[0]
+
+
+def check_object(db, table, key, expected_attributes):
+    tbl =  swsscommon.Table(db, table)
+    keys = tbl.getKeys()
+    assert key in keys, "The desired key is not presented"
+
+    status, fvs = tbl.get(key)
+    assert status, "Got an error when get a key"
+
+    assert len(fvs) >= len(expected_attributes), "Incorrect attributes"
+
+    attr_keys = {entry[0] for entry in fvs}
+
+    for name, value in fvs:
+        if name in expected_attributes:
+            assert expected_attributes[name] == value, "Wrong value %s for the attribute %s = %s" % \
+                                               (value, name, expected_attributes[name])
+
+def get_key_with_attr(db, table, expected_attributes ):
+    tbl =  swsscommon.Table(db, table)
+    keys = tbl.getKeys()
+    retkey = list()
+    #assert key in keys, "The desired key is not presented"
+
+    for key in keys:
+      status, fvs = tbl.get(key)
+      assert status, "Got an error when get a key"
+
+      assert len(fvs) >= len(expected_attributes), "Incorrect attributes"
+
+      attr_keys = {entry[0] for entry in fvs}
+
+      num_match = 0
+      for name, value in fvs:
+          if name in expected_attributes:
+                 if expected_attributes[name] == value:
+                    num_match += 1
+      if num_match == len(expected_attributes):
+         retkey.append(key)
+
+    return retkey
+       
+      
+def create_evpn_nvo(dvs, nvoname, tnlname):
+    conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+    attrs = [
+            ("source_vtep", tnlname),
+    ]
+
+    # create the VXLAN tunnel Term entry in Config DB
+    create_entry_tbl(
+        conf_db,
+        "VXLAN_EVPN_NVO", '|', nvoname,
+        attrs,
+    )
+
+def remove_evpn_nvo(dvs, nvoname):
+    conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+    delete_entry_tbl(conf_db,"VXLAN_EVPN_NVO", nvoname,)
+
+def create_vxlan_tunnel(dvs, name, src_ip, dst_ip = '0.0.0.0', skip_dst_ip=True):
+    conf_db = swsscommon.DBConnector(swsscommon.CONFIG_DB, dvs.redis_sock, 0)
+
+    attrs = [
+            ("src_ip", src_ip),
+    ]
+
+    if not skip_dst_ip:
+        attrs.append(("dst_ip", dst_ip))
+
+    # create the VXLAN tunnel Term entry in Config DB
+    create_entry_tbl(
+        conf_db,
+        "VXLAN_TUNNEL", '|', name,
+        attrs,
+    )
+
+def create_vxlan_tunnel_map(dvs, tnlname, mapname, vni_id, vlan_id):
```

Comment:
general comment - variables can be followed either map_name or mapName.

Reply:
Done